kuronometer-core/src/main/scala/com/github/pedrovgs/kuronometer/free/interpreter/csv/CsvReporter.scala

```diff
@@ -0,0 +1,99 @@
+package com.github.pedrovgs.kuronometer.free.interpreter.csv
+
+import java.io.{File, FileReader, FileWriter}
+
+import com.github.pedrovgs.kuronometer.KuronometerResults.KuronometerResult
+import com.github.pedrovgs.kuronometer.free.domain._
+import org.supercsv.io.{CsvBeanReader, CsvBeanWriter, ICsvBeanWriter}
+import org.supercsv.prefs.CsvPreference
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+class CsvReporter(implicit csvReporterConfig: CsvReporterConfig) {
+
+  def report(buildExecution: BuildExecution): KuronometerResult[BuildExecution] = {
+    val stages = buildExecution.buildStagesExecution.stages
+    writeBuildStages(stages)
+    Right(buildExecution)
+  }
+
+  def getTotalBuildExecutionStages: KuronometerResult[SummaryBuildStagesExecution] = {
+    getBuildExecutionStagesSinceTimestamp(0)
+  }
+
+  def getBuildExecutionStagesSinceTimestamp(filterTimestamp: Long): KuronometerResult[SummaryBuildStagesExecution] = {
+    if (!existsReportFile || reportFileIsEmpty) {
+      Right(SummaryBuildStagesExecution())
+    } else {
+
+      val csvBuildStages = mutable.ListBuffer[CsvBuildStageExecution]()
+
+      val csvFile = new File(csvReporterConfig.executionTasksCsvFile)
+      val beanReader = new CsvBeanReader(new FileReader(csvFile), CsvPreference.STANDARD_PREFERENCE)
+      val headers = beanReader.getHeader(true)
+      try {
+        var csvBuildStage: CsvBuildStageExecution = beanReader.read(classOf[CsvBuildStageExecution], headers, CsvReporterConfig.processors: _*)
+        while (csvBuildStage != null) {
+          if (csvBuildStage.timestamp >= filterTimestamp) {
+            csvBuildStages.append(csvBuildStage)
+          }
+          csvBuildStage = beanReader.read(classOf[CsvBuildStageExecution], headers, CsvReporterConfig.processors: _*)
+        }
+      } finally if (beanReader != null) {
+        beanReader.close()
+      }
+      Right(mapCsvBuildStages(csvBuildStages))
+    }
+  }
+
+  def mapCsvBuildStages(csvBuildStages: ListBuffer[CsvBuildStageExecution]): SummaryBuildStagesExecution = {
+    val stages = csvBuildStages.map { stage =>
+      SummaryBuildStageExecution(stage.name, stage.executionTime, stage.timestamp)
+    }
+    SummaryBuildStagesExecution(stages)
+  }
+
+  def clear(): Unit = {
```

Comment:
As this method was being used just for testing purposes I've implemented this functionality from my tests instead of using this impure function. Thanks!